packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -385,11 +387,23 @@ class _ModalScopeStatus extends InheritedWidget {
   }
 }
 
+/// Indicates whether the current route should be popped.
+enum RoutePopDisposition {
+  /// Pop the route.
+  pop,
+
+  /// Do not pop the route.
+  doNotPop,
+
+  /// Delegate this to the next level of navigation.
+  bubble,
+}
+
 /// Signature for a callback that verifies that it's OK to call [Navigator.pop].
 ///
 /// Used by [Form.onWillPop], [ModalRoute.addScopedWillPopCallback], and
 /// [ModalRoute.removeScopedWillPopCallback].
-typedef Future<bool> WillPopCallback();
+typedef Future<RoutePopDisposition> WillPopCallback();
```

Comment:
I wonder if we should leave the callback returning a Boolean.  It's not obvious to me that returning "bubble" means "keep calling the rest of the callbacks".